src/blt_mad/helpers.go

```diff
@@ -146,26 +134,96 @@ func withinTolerance(a float64, b float64, e float64) bool {
 	}
 }
 
-func withinToleranceFloatSlice(a []float64, b []float64, e float64) bool {
+func WithinToleranceFloatSlice(a []float64, b []float64, e float64) bool {
 	if reflect.DeepEqual(a, b) {
 		return true
 	} else {
 		for i := 0; i < len(a); i++ {
-			if !withinTolerance(a[i], b[i], e) {
+			if !WithinTolerance(a[i], b[i], e) {
 				return false
 			}
 		}
 	}
 	return true
 }
 
-func readTxtToString(filePath string) (string, error) {
-	// Read the entire file into a byte slice
-	content, err := ioutil.ReadFile(filePath)
+func calculatePercentile(numbers []float64, percentile float64) float64 {
+	sort.Float64s(numbers)
+	index := int(percentile / 100 * float64(len(numbers)-1)) //index corresponding to the percentile
+	lower := numbers[index]
+	upper := numbers[index+1]
+	fractionalPart := percentile/100*float64(len(numbers)-1) - float64(index)
+	value := lower + (upper-lower)*fractionalPart
+	return value
+}
+
+func GetValuesLargerThanPercentile(numbers []float64, percentile float64) []float64 {
+	valueAtPercentile := calculatePercentile(numbers, percentile)
```

Comment:
The reason I asked the previous question is that sorting the input by the function that is called here could potentially impact the logic of this function: for example, in the loop below, you are filtering numbers that are greater than `valueAtPercentile` from a list that is already sorted. So the algorithm could've been implemented in a more efficient way. And in general, once you sort one of these lists, you wouldn't need to sort them again, let's say in a different function.

Reply:
I fixed the sorting issue from the previous comment but I will rerun getting testing data to make sure that it did not get messed up by that issue